codec/reflectcodec/struct_fielder.go

```diff
@@ -114,3 +110,11 @@ func (s *structFielder) GetSerializedFields(t reflect.Type) ([]FieldDesc, error)
 	s.serializedFieldIndices[t] = serializedFields // cache result
 	return serializedFields, nil
 }
+
+func (s *structFielder) getCachedSerializedFields(t reflect.Type) ([]FieldDesc, bool) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	cachedFields, ok := s.serializedFieldIndices[t]
+	return cachedFields, ok
+}
```

Comment:
I waffled back and forth on if this should be it's own function... but ended up making it it's own function so that we could defer the `RUnlock`